### DISCUSSION TITLE
1.23.0

### DISCUSSION THREAD

**Discussion by Koenkk** [Koenkk/zigbee2mqtt#11170] [General]
# Notes
- Due to #10437 there were some internal changes, this can impact your custom user extension. ([changeset](https://github.com/Koenkk/zigbee2mqtt/commit/30177b0db403ae5998013c603da96aad8f59ad9a))

# Features
- #10141 Local OTA index ([documentation](https://www.zigbee2mqtt.io/guide/usage/ota_updates.html))

# Improvements
- #10133 Automatically expose to HA a entitity as energy when Wh/kWh exists 
- [#3600](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3600) Support detection_interval for Xiaomi RTCGQ13LM
- #9493 Poll state (on/off) for some TS011F_plug_3 types
- [#3622](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3622) Expose status for ShinaSystem CSM-300ZB
- [#3627](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3627) Expose temperature_setpoint_hold for Centralite 3157100-E
- #10555 Log out of memory error when initial build fails
- #5706 Support illuminance for Xiaomi RTCGQ12LM
- #9057 Enable reporting of physical on/off and energy changes for TS011F_plug_3
- [#3640](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3640) Added support for setting min\max level of brightness BTicino L441C/N4411C/NT4411C and Legrand 067797
- [#3633](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3633) Support changing per device scene states
- [#3649](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3649) Support setting TuYa LCZ030 min/max temperature/humidity alarm
- #10672 Support `homeassistant_rename` for groups
- #10437 Restructure settings (not a breaking change)
- [#3660](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3660) Support more actions and battery for Tuya Smart Knob ERS-10TZBVK-AA
- [#3665](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3665) Support pin add/delete for Danalock V3-BTZB/V3-BTZBE
- #5137 Add power, voltage and current calibration option
- [#85](https://github.com/Koenkk/zigbee-OTA/pull/85) Support OTA for Legrand 067646, 064888 and 067775/741811
- [#488](https://github.com/Koenkk/zigbee-herdsman/pull/488) Improve handling of greenpower devices on larger networks
- [#3719](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3719) Support OTA updates for Xiaomi VOCKQJK11LM
- [#3732](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3723) Add additional functionality for WB-MSW-ZIGBEE v.3
- [#3729](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3729) Add OTA for Xiaomi RTCGQ12LM and RTCGQ13LM
- [#3762](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3762) [#3779](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3779) [#3791](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3791) Support `flip_indicator_light` for Xiaomi H1 and E1 switches
- [#3757](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3757) Improve ZNJLBL01LM support
- [#3773](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3773) Add RGBW mode for Aqara ZNDDMK11LM 
- [#3780](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3780) Support `power_outage_memory` for ZNDDMK11LM
- [#3775](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3775) Allow to send ZCL commands via MQTT and the frontend
- [#3781](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3781) Support temperature and battery for Sinope WL4200 and WL4200S, remove tamper
- #11107 Allow to override HA entity/device name

# Fixes
- [#3599](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3599) Fix max/min temperature for TuYa TS0601_thermostat
- [#3604](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3604) Remove temperature from Xiaomi MCCGQ14LM
- #10590 Fix deadzone_temperature range and step size
- #10471 Fix no occupancy false being send for Xiaomi RTCGQ13LM
- #10497 Fix HGZB-42 being recognised as NUE-AUWZO2
- #10489 Fix time synchronization not working for TuYa devices
- #10462 Fix NodOn SIN-4-RS-20 reporting every few seconds
- #10504 #10930 Fix arm action not working for 3400-D, KEYZB-110, KEYPAD001, ZS130000078, XHK1-TC, XHK1-UE
- #10245 Remove humidity from THS317-ET (device does not support it)
- [#3617](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3617) Fix exposes for BTicino K4003C/L4003C/N4003C/NT4003C and L441C/N4411C/NT4411C
- #10605 Fix 33943/33944/33946 recognised as 33951/33948
- #10610 Force QBKG23LM to be a mains powered router
- [#3637](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3637) Force IL06_1 power source to battery
- [#3644](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3644) Fix LCZ030 not reporting measurements
- #10656 Fix Xiaomi router devices discovered as another Xiaomi devices 
- [#3656](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3656) Fix `action_color` calculation
- [#3655](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3655) Fix actions missing in expose of RGB Genie ZB-5028
- [#3657](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3657) Update TS0601_thermostat local temperature calibration
- #10299 Fix `payload is not iterable` exception when configuring reporting
- #10709 Fix electrical measurements for Xiaomi SSM-U01 inconsistent
- #10775 Fix ZNLDP12LM color temperature range
- #10666 Fix update stuck at 100%
- #10131 Fix device softwareBuildID and dateCode not refreshed after OTA update
- [#3218](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3218) LED1624G9 doesn't support color temperature
- #10619 Fix thermostat control failing with numbers having decimals
- #10782 Fix unable to pair green power devices via routers when using the Conbee adapter
- #5244 Clear old availability topic when device is renamed
- #10348 Fix NAS-AB02B2 alarm volume inverted
- #10972 #10862 Fix no action for Legrand 067773 and 067774
- #9561 Fix power measurements not working for PLUG EDP RE:DY
- #10905 Fix log files missing end of logging
- [#3746](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3746) Change ZNCZ04LM max overload_protection to 2300 Watt
- #11028 Fix PECMS01 battery percentage incorrect
- [#3754](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3754) Fix duplicate messages for E220-KR4N0Z0-HA
- #7405 TS011F_plug `_TZ3000_v1pdxuqq` doesn't support power measurements 
- #11126 Remove state property for Xiaomi WXCJKG11LM, WXCJKG12LM and WXCJKG13LM
- [#1191](https://github.com/Koenkk/zigbee2mqtt.io/pull/1191) Add measurement_poll_interval option to TuYa TS0121_plug
- [#3789](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3789) Fix device configure for TuYa YSR-MINI-Z and ERS-10TZBVK-AA

# New supported devices
This release adds support for **86** new devices:
- `TS0601_smart_air_house_keeper` TuYa Smart air house keeper
- `064882` Legrand Cable outlet with pilot wire and consumption measurement
- `12501` Scan Products Zigbee push dimmer
- `14156506L` Lidl Livarno Lux smart LED mood light
- `1746630V7` Philips Amarant linear outdoor light
- `3261330P6` Philips Hue white ambiance Still
- `3417511P9` Philips Hue white ambiance bathroom recessed downlight Adore with Bluetooth
- `3802960` Namron LED 9W DIM E27
- `3802963` Namron LED 5,3W DIM E14
- `3802965` Namron LED 4,8W DIM GU10
- `3802966` Namron LED 4.8W CCT GU10
- `4023330P6` Philips Hue white ambiance suspension Amaze
- `4080248P7` Philips Hue Signe floor light
- `41E2PBSWMZ/356PB2MBTZ` Schneider Electric Wiser 40/300-Series module switch 2A
- `43096` Enbrighten Zigbee plug-in smart dimmer with dual controlled outlets
- `43132` Jasco Zigbee smart outlet
- `4512733` Namron ZigBee dimmer 2-pol 400W
- `5412748727364` Prolight E27 bulb dimmable
- `545D6115` Schneider Electric LK FUGA wiser wireless socket outlet
- `545D6306` Schneider Electric LK FUGA Wiser wireless PIR with relay
- `915005996401` Philips Hue white ambiance ceiling light Enrave S with Bluetooth
- `915005997501` Philips Hue Bluetooth white & color ambiance ceiling lamp Infuse large
- `9290019533` Philips Hue white ambiance GU10 with Bluetooth
- `9290022415` Philips Hue White G25 E26 Edison Filament Globe Bluetooth
- `9290024691` Philips Hue white single filament bulb A19 E26 with Bluetooth (1100 Lumen)
- `9290024693` Philips Hue white A60 bulb B22 1055lm with Bluetooth
- `929002994901` Philips Hue gradient lightstrip
- `929003045301` Philips Hue White and Color Ambiance GU10 (Centura)
- `929003047501` Philips Centura recessed spotlight
- `929003055801` Philips Hue white ambiance bathroom ceiling light Adore with Bluetooth
- `929003055901` Philips Hue white ambiance bathroom ceiling light Adore with Bluetooth
- `929003056901` Philips Hue Struana 27W
- `AC33898` LEDVANCE  Smart+ LED classic E27 Zigbee 10W/810lm
- `B1027EB0Z01` LG Electronics Smart bulb 1
- `B1027EB0Z02` LG Electronics Smart bulb 2
- `CM10ZW` ORVIBO Multi-functional 3 gang relay
- `CSLC601-D-E` CASAIA Dry contact relay switch module in 220v AC for gas boiler
- `E5` Nous Temperature & humidity
- `ED-10042` EcoDim Zigbee LED filament light dimmable E27, globe G125, flame 2200K
- `EFEKTA_PWS_MaxPro` Custom devices (DiY) [Plant watering sensor EFEKTA PWS Max Pro,  long battery life](http://efektalab.com/PWS_MaxPro)
- `EFEKTA_eON29wz` Custom devices (DiY) [Mini weather station, barometer, forecast, charts, temperature, humidity, light](http://efektalab.com/eON290wz)
- `EZ200` Evanell Thermostatic radiator valve
- `GL-B-003P` Gledopto Zigbee 7W E26/E27 Bulb RGB+CCT (pro)
- `GL-C-003P` Gledopto Zigbee LED Controller RGB (pro)
- `JT-BZ-01AQ/A` Xiaomi Aqara smart natural gas detector
- `KK-ES-J01W` Konke Room temperature, relative humidity and illuminance sensor
- `LDSENK01F` ADEO 10A EU smart plug
- `MIR-HE200-TY_fall` TuYa Human presence sensor with fall function
- `MS100` TP-Link Smart motion sensor
- `NAS-AB02B2` Neo Alarm
- `PJ-ZGD01` TuYa Garage door opener
- `PMM-300Z2` ShinaSystem SiHAS energy monitor
- `RB 251 C` Innr E14 bulb RGBW
- `RTCZCGQ11LM` Xiaomi Aqara Presence Detector FP1 (regions not supported for now)
- `SIN-4-RS-20` NodOn Roller shutter controller
- `SM-0306E-2W` UseeLink 4 gang switch, with USB
- `SZ-ESW02` Sercomm Telstra smart plug 2
- `TH008L10RGBCCT` UR Lighting 10W RGB+CCT downlight
- `TS0501A` TuYa Zigbee light
- `TS0601_smart_air_house_keeper` TuYa Smart air house keeper
- `TS0601_vibration_sensor` TuYa Smart vibration sensor
- `TS110E_1gang` Lonsonho Zigbee smart dimmer module 1 gang with neutral
- `TS110E_2gang` Lonsonho Zigbee smart dimmer module 2 gang with neutral
- `TS130F_dual` Lonsonho Dual curtain/blind module
- `UP-SA-9127D` Sunricher LED-trading 2 channel AC switch
- `WLS-100z` TuYa Water leak sensor
- `WXKG14LM` Xiaomi Aqara wireless remote switch H1 (single rocker)
- `WZ5_cct` Skydance Zigbee & RF 5 in 1 LED controller (CCT mode)
- `WZ5_dim` Skydance Zigbee & RF 5 in 1 LED controller (DIM mode)
- `WZ5_rgb` Skydance Zigbee & RF 5 in 1 LED controller (RGB mode)
- `WZ5_rgbcct` Skydance Zigbee & RF 5 in 1 LED controller (RGB+CCT mode)
- `WZ5_rgbw` Skydance Zigbee & RF 5 in 1 LED controller (RGBW mode)
- `XFY-CGQ-ZIGB` TuYa Illuminance sensor
- `ZB250` Micro Matic Norge AS Zigbee dimmer for LED
- `ZB3102` Jasco Products Zigbee plug-in smart dimmer
- `ZIGBEE-B09-UK` Zemismart Zigbee smart outlet universal socket with USB port
- `ZM-35H-Q` TuYa Motion Sensor
- `ZM-AM02` Zemismart Zigbee/RF curtain converter
- `ZNLDP13LM` Xiaomi Aqara T1 smart LED bulb
- `ZNXDD01LM` Xiaomi Aqara ceiling light L1-350
- `ZP-LZ-FR2U` Moes Zigbee 3.0 dual USB wireless socket plug
- `ZPLUG_Boost` CLEODE ZPlug boost
- `iL03_1` Iris Smart plug
- `tubeszb.router` TubesZB [CC2652 Router](https://github.com/tube0013/tube_gateways/tree/main/tube_cc_router)
- `ALCANTARA2` Acova Alcantara 2 heater
- `R7048` Woox Smart humidity & temperature sensor

<hr /><em>This discussion was created from the release <a href='https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.23.0'>1.23.0</a>.</em>

**Comment by jeroenvdd** [comment]
![image](https://user-images.githubusercontent.com/6975276/152032285-62d0bd79-64fa-4504-95d2-d8f6ab7bfa23.png)
Battery state seems to be ok now!

**Comment by mgiebels** [comment]
After update i get 502: Bad Gateway

I did a roll back and everything is working again!

**Reply by robertalexa** [reply]
You need to wait for the frontend to completely load, it can take a couple of minutes :)

**Comment by Tonio16** [comment]
Hello


I installed the 1.23 and tried the ota for legrand but it doesn’t work:
Controller is a sonoff.
```
Version Zigbee2MQTT
1.23.0 commit: 6ea61bae
Type de Coordinateur
zStack3x0
Révision du Coordinateur
20211217
Adresse IEEE du Coordinateur
0x00124b00258d7e3c
Version Interface
0.6.71
```

`Is new image available for '0x00047400000db744', current '{"fieldControl":0,"manufacturerCode":4129,"imageType":16,"fileVersion":1851907}', latest meta '{"fileVersion":2966019,"fileSize":242695,"url":"https://github.com/Koenkk/zigbee-OTA/raw/master/images/Legrand/1021-0010-2d4203-NLM-45.zigbee","sha512":"fc3a7f49cef2a6847168cfe165b335feeecbcd64c8729afa4b6aea302988ebf5a17c85b285230ece7c9cc469017e8b694f5ef33481bdfce8d5fb9b5645f98c70"}'
Debug Received Zigbee message from 'Module dressing', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":1851907,"imageType":16,"manufacturerCode":4129}' from endpoint 1 with groupID 0
Debug Device 'Module dressing' requested OTA
Debug Received Zigbee message from 'Module dressing', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":1851907,"imageType":16,"manufacturerCode":4129}' from endpoint 1 with groupID 0
Debug Device 'Module dressing' requested OTA
Debug Received Zigbee message from 'Module dressing', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":1851907,"imageType":16,"manufacturerCode":4129}' from endpoint 1 with groupID 0
Debug Device 'Module dressing' requested OTA
Debug Received MQTT message on 'zigbee2mqtt/bridge/request/device/ota_update/update' with data '{"id":"Module dressing","transaction":"5zsfw-13"}'
Debug Update of 'Module dressing' failed (Error: Device didn't respond to OTA request)
Info MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/update', payload '{"data":{"id":"Module dressing"},"error":"Update of 'Module dressing' failed (Device didn't respond to OTA request)","status":"error","transaction":"5zsfw-13"}'
Error Update of 'Module dressing' failed (Device didn't respond to OTA request)
Debug Error: Device didn't respond to OTA request at requestOTA (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/lib/ota/common.js:196:15) at Object.updateToLatest (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/lib/ota/common.js:283:21) at OTAUpdate.onMQTTMessage (/opt/zigbee2mqtt/lib/extension/otaUpdate.ts:256:21)`

Antoine

**Comment by itavero** [comment]
Just noticed that the Zemismart ZM-AM02 is not exposed as a `cover`, but I think it actually is one. Any specific reason for that, @Burki24?

**Reply by Koenkk** [reply]
https://github.com/Koenkk/zigbee2mqtt/issues/11190

**Reply by Burki24** [reply]
The ZM-AM02 isn't a normal cover. It's a dongle to pair with different covers. So it's not correct, to place it in tuya-cover. It is now placed in Zemismart as own Device. It can be used with different RF-covers. And has more possibilites to adjust the paired cover:
![image](https://user-images.githubusercontent.com/20765982/152584440-606c0d36-6a5f-43ca-9fa8-b91593072fc8.png)

Actually, I'm working on an update for the converter to make all DPs usable.

**Comment by OrionExplorer** [comment]
Hey @Koenkk, thank you for the release! Am I wrong, or the [Supported Devices](https://www.zigbee2mqtt.io/supported-devices/) have not been updated yet?

**Reply by Koenkk** [reply]
whoos forgot to merge develop to master, should be up to date in an hour from now.

**Reply by OrionExplorer** [reply]
Fantastic, thanks!

**Comment by mgiebels** [comment]
I have updated again but web ui gives again bad gateway 

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] socat.sh: executing... 
[17:51:34] INFO: Socat not enabled, marking service as down
[cont-init.d] socat.sh: exited 0.
[cont-init.d] zigbee2mqtt.sh: executing... 
[17:51:34] INFO: MQTT available, fetching server detail ...
[17:51:34] INFO: Previous config file found, checking backup
[17:51:34] INFO: Creating backup config in '/share/zigbee2mqtt/.configuration.yaml.bk'
[17:51:35] INFO: Adjusting Zigbee2mqtt core yaml config with add-on quirks ...
[cont-init.d] zigbee2mqtt.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[17:51:35] INFO: Handing over control to Zigbee2mqtt Core ...
> zigbee2mqtt@1.23.0 start
> node index.js
CREATED DECONZ ADAPTER


My config:

data_path: /share/zigbee2mqtt
external_converters: []
devices: devices.yaml
groups: groups.yaml
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://core-mosquitto:1883
  user: mqtt
  password: xxxx
serial:
  port: >-
    /dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2225559-if00
  adapter: deconz
advanced:
  log_level: warn
  pan_id: 6754
  channel: 11
  network_key:
    - 1
    - 3
    - 5
    - 7
    - 9
    - 11
    - 13
    - 15
    - 0
    - 2
    - 4
    - 6
    - 8
    - 10
    - 12
    - 13
  availability_blocklist: []
  availability_passlist: []
  ikea_ota_use_test_url: true
  lpog_level: info
device_options: {}
blocklist: []
passlist: []
queue: {}
frontend:
  port: 8099
experimental:
  new_api: true
availability: false
socat:
  enabled: false
  master: pty,raw,echo=0,link=/dev/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: '-d -d'
  log: false
  restartdelay: 1
  initialdelay: 1


When i rollback to 1.22 de web ui works fine

**Comment by thanasiskalos** [comment]
Hello 
Upon update to 1.23.-1 on Home Assistant I lost the Temp on all  MCCGQ14LM which was introduced on the last version.

Also I lost the connection to MQTT for the one  of the ZNCZ04LM , though the others were fine. The strange is that from ZIGbee2Mqtt web interface I could control that specific   ZNCZ04LM  but there was no communication with MQTT broker. 
 
I had to restore version 1.22.2-1 from backup to have all functionality back.

Regards

**Comment by RHumyn** [comment]
Hey guys. 
Upon an update to 1.23.0 on Home Assistant, all devices appear without native pictures. I had a 502 error before but fixed it by turning on/of the stick a few times.
Do you have any idea why I can't get pictures? Also, do you have functionality for a force updating all device's pictures?  
<img width="233" alt="Screenshot 2022-02-06 at 19 55 24" src="https://user-images.githubusercontent.com/25850760/152694673-7aa2be32-31d6-4011-b4f4-e1d41e2d73f8.png">
<img width="245" alt="Screenshot 2022-02-06 at 19 55 38" src="https://user-images.githubusercontent.com/25850760/152694677-80d9f2e0-416d-4c6c-ac6b-540ee6e60472.png">

**Comment by Loic691** [comment]
Hi. On 1.33.1 the OTA not working for update 067646 Home away Legrand battery switch.
The update is found but nothing when we click on update device. Even if we click button on the device...

**Reply by Tonio16** [reply]
Hello

Did you try to put the battery out/in before the update?


Antoine

**Reply by Loic691** [reply]
Oh no sorry
But the ota update of this devis seams to be ok from 1.23...

**Reply by Loic691** [reply]
> Hello
> 
> Did you try to put the battery out/in before the update?
> 
> 
> Antoine

No
Do you have documentation on this ?

**Reply by Tonio16** [reply]
No, but for the one on 230V, you have to.

Antoine

**Reply by Loic691** [reply]
The 230V i have have ota ok
My problem is just on battery device....

**Comment by lux73** [comment]
@Loic691 did you realise that you Post in the 1.23.0 Thread from 02/01/2022?
